// Copyright 2017 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
////////////////////////////////////////////////////////////////////////////////

syntax = "proto3";

package google.cloud.crypto.tink;

option java_package = "com.google.cloud.crypto.tink";
option java_outer_classname = "AesGcmProto";

// only allowing IV size in bytes = 12 and tag size in bytes = 16
// Thus, accept no params.
message AesGcmParams {
}

message AesGcmKeyFormat {
  AesGcmParams params = 1;
  int32 key_size = 2;
}

// key_type: type.googleapis.com/google.cloud.crypto.tink.AesGcmKey
message AesGcmKey {
  int32 version = 1;
  AesGcmParams params = 2;
  bytes key_value = 3;
}
